### ISSUE TITLE
Fix `declaration-property-value-no-unknown` false positives for `math` of `font-size`

### ISSUE THREAD

**Issue by otomad** [stylelint/stylelint#8492]
### What minimal example or steps are needed to reproduce the bug?

```css
math {
	font: math math;
	font-size: math;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
    "declaration-property-value-no-unknown": true
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIC2BDALgCwATADoDtc0AzCfNeLVTK9DAbgKNPIFoBnASwC8ZLqGBAL4EQAGhDFOAGxgA5FEjiIBAOjDt248GSkBzBCDz4sWXCBgAPNDHxR25ygG0Cp0+fZoAnrOmc2kPj6HGgodigATlDmrlgAusLagfoAYhAR1IYAVuxk2rAADlqIxu4gnj4wfuSOZiDS6DCe5mKxHt6+-misyZx6IWFQkdEIdQ02zSCJQkA)

### Which Stylelint-related dependencies are you using?

```json
{
    "stylelint": "16.16.0"
}
```

### What did you expect to happen?

No problems to be reported.

### What actually happened?

The following problems were reported:

```
test.css
2:8-12    ✖  Unexpected unknown value "math" for property "font" (declaration-property-value-no-unknown)
3:13-17   ✖  Unexpected unknown value "math" for property "font-size" (declaration-property-value-no-unknown)
```

### Do you have a proposal to fix the bug?

This issue is used to split pr #8489.

Related: #8436.

**Comment by skyclouds2001** [comment]
see https://github.com/mdn/data/pull/945
